barcodecop/barcodecop.py

```diff
@@ -272,6 +270,13 @@ def main(arguments=None):
     if args.qual_filter:
         filtered = ifilterfun(qual_filter, filtered)
 
+    if args.read_counts:
+        filtered, filtered2 = tee(filtered)
+        input_count = len(list(seqs2))
+        output_count = len(list(filtered2))
```

Comment:
A possible problem here is that forcing these generators to lists will create a lot of memory overhead (I think that otherwise we manage to do all of the filtering without reading the whole file into memory). One alternative is to use a counter (or ``enumerate()``) to get a tally of filtered reads. Then the question is how to tally the input without reading the whole thing into a list.

Reply:
https://github.com/nhoffman/barcodecop/pull/11/commits/d4bac235398b52ebbaf5fdde6dc9756b8f0c5c12